components/page_wrappers/TranslationsWrapper.tsx

```diff
@@ -1,15 +1,19 @@
 import React from 'react';
 import { SupportedLocale } from '../../@types';
+import { LanguageSpecificTranslation, translations } from '../../utils/translations';
 
 type Props = {
   locale: SupportedLocale
   children: JSX.Element
 };
 
-export const TranslationContext = React.createContext<{ locale: SupportedLocale }>({ locale: 'en' });
+export const TranslationContext = React.createContext<{
+  locale: SupportedLocale, translation: LanguageSpecificTranslation
+}>({ locale: 'en', translation: translations.en });
+
 export const TranslationsWrapper = ({ locale, children }: Props) => {  
   return (
-    <TranslationContext.Provider value={ { locale } } >
+    <TranslationContext.Provider value={ { locale, translation: translations[locale] } } >
```

Comment:
This whole thing is smart. Bravo!

![](https://media3.giphy.com/media/IjaIGVkCsky3lq77by/giphy.gif)

Reply:
Thank you~